core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -123,11 +123,10 @@ public AbstractClient(ClientContext context, InetSocketAddress address,
   protected abstract ServiceType getRemoteServiceType();
 
   protected long getRemoteServiceVersion() throws AlluxioStatusException {
-    return retryRPC(() ->
```

Comment:
Are you saying there is a retry of `getRemoteServiceVersion` outside of this, so we don't need to use `retryRPC`? I don't understand where the infinite loop is happening.

Reply:
When a channel creation succeeds but `getRemoteServiceVersion()` fails always.
In that case, `retryRpcInternal` will call disconnect after the rpc, then connect again. After disconnect, connect will call `getRemoteServiceVersion()` again.

Reply:
This will create connect -> getRemoteServiceVersion -> connect -> getRemoteServiceVersion loop. `connect` after `getRemoveServiceVersion` being the second attempts.

Reply:
I see. Then, could you add a comment here saying we don't need to retry here, since the the entire connection/channel will be retried outside?